### ISSUE TITLE
Cannot use a scalar value as an array options.php

### ISSUE THREAD

**Issue by OsoBear** [tollmanz/wordpress-pecl-memcached-object-cache#27]
After installing the drop in, I get the error:
Warning: Cannot use a scalar value as an array in /var/www/vhosts/domain.com/httpdocs/wp-includes/option.php on line 68

Any ideas?
WP  Version 3.8.1

**Comment by tollmanz** [comment]
That is an odd error. I have a few questions to see if we can find the issue:
1. Do you have [PECL Memcached](http://pecl.php.net/package/memcached) installed? Note that this is different from [PECL Memcache](http://pecl.php.net/package/memcache).
2. If you run this PHP script on your server, what results do you get?
   
   ``` php
   error_reporting(E_ALL & ~E_NOTICE); 
   
   $mc = new Memcached(); 
   $mc->addServer("localhost", 11211); 
   
   $mc->set("foo", "Hello!"); 
   $mc->set("bar", "Memcached..."); 
   
   $arr = array( 
       $mc->get("foo"), 
       $mc->get("bar") 
   ); 
   var_dump($arr); 
   
   var_dump($mc->getStats());
   ```
3. Can you do a full stack trace on the issue, perhaps with [Xdebug](http://xdebug.org/)?

My hunch is that you do not have Memcached properly configured. The line referenced is:

``` php
$notoptions[$option] = true;
```

Earlier, `$notoption` is set with:

``` php
$notoptions = wp_cache_get( 'notoptions', 'options' );
```

It's possible that this is returning an unexpected value (something that is scalar) and thus the error is thrown. The only way I could imagine this happening is if the Memcached setup is incorrectly.

**Comment by tollmanz** [comment]
Hi @OsoBear! Any updates on this? If I do not hear back from you soon, I will close this ticket. I'm happy to help you in the future, but I just want to keep the issues cleaned up.

**Comment by OsoBear** [comment]
I went ahead and have been using Ryan's version, but I'll give your suggestion a try to see if that works. Hopefully later this weekend. To answer your question - I have both PECL Memcached and PECL Memcache installed. I'll reinstall your version and try running your PHP script and see if that works. One other side note -- I'm using PHP-FPM 5.5. I'm assuming your memcached backend is compatible?

thanks!
elaine

**Comment by tollmanz** [comment]
Thanks for the feedback @OsoBear! Boren's Memcached backend is definitely the champion. It has been in production for a LONG time. I appreciate that you gave this a shot. 

I do not know how well this will run with PHP 5.5 as I do not have any WordPress sites running 5.5 in production. I've done some dev testing in local development boxes and did not notice any issues.

Since you opened this issue, I finally added unit tests for the object cache. I wrote these 1.5 years ago, but never added them to the repo. Your question prompted me to run them against PHP 5.5, which passes.

I definitely would appreciate any more effort on your part to get to the bottom of the error. I know you probably don't have time to chase it down, but I would love it if you did.

Thanks!
